test/e2e/e2e_v1.go

```diff
@@ -0,0 +1,147 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/kubernetes-sigs/kubebuilder/test/e2e/framework"
+	e2einternal "github.com/kubernetes-sigs/kubebuilder/test/internal/e2e"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("v1 main workflow", func() {
+	It("should perform main kubebuilder workflow successfully", func() {
+		testSuffix := framework.RandomSuffix()
+		c := initConfig(testSuffix)
+		kubebuilderTest := e2einternal.NewKubebuilderTest(c.workDir, framework.TestContext.BinariesDir)
+
+		prepare(c.workDir)
+		defer cleanupv1(kubebuilderTest, c.workDir, c.controllerImageName)
+
+		var controllerPodName string
+
+		output, errrr := exec.Command("which", "kubebuilder").Output()
+		fmt.Printf("output of which: %s", output)
+		Expect(errrr).NotTo(HaveOccurred())
+
+		By("init v1 project")
+		initOptions := []string{
+			"--project-version", "v1",
+			"--domain", c.domain,
+			"--dep", "true",
+		}
+		err := kubebuilderTest.Init(initOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating api definition")
+		crdOptions := []string{
+			"--group", c.group,
+			"--version", c.version,
+			"--kind", c.kind,
+			"--namespaced", "false",
+			"--resource", "true",
+			"--controller", "true",
+		}
+		err = kubebuilderTest.CreateAPI(crdOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		// TODO: enable this test after we support gen rbac for core type controller
+		By("creating core-type resource controller")
+		coreControllerOptions := []string{
+			"--group", "apps",
+			"--version", "v1",
+			"--kind", "Deployment",
+			"--namespaced", "true",
+			"--resource", "false",
+			"--controller", "true",
+		}
+		err = kubebuilderTest.CreateAPI(coreControllerOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("building image")
+		makeDockerBuildOptions := []string{"docker-build"}
+		err = kubebuilderTest.Make(makeDockerBuildOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("pushing image")
+		makeDockerPushOptions := []string{"docker-push"}
+		err = kubebuilderTest.Make(makeDockerPushOptions)
+		Expect(err).NotTo(HaveOccurred())
+
+		// NOTE: If you want to run the test against a GKE cluster, you will need to grant yourself permission.
+		// Otherwise, you may see "... is forbidden: attempt to grant extra privileges"
+		// $ kubectl create clusterrolebinding myname-cluster-admin-binding --clusterrole=cluster-admin --user=myname@mycompany.com
+		// https://cloud.google.com/kubernetes-engine/docs/how-to/role-based-access-control
+		By("installing CRD and RBAC")
+		makeInstallOptions := []string{"install"}
+		err = kubebuilderTest.Make(makeInstallOptions)
```

Comment:
We don't need to invoke `install` now because `deploy` does that for us.

Reply:
You are right.
After looking at the Makefile again, I guess we can make `deploy` target invoke `install` target first.